lib/Command/PointRestore.php

```diff
@@ -907,4 +908,28 @@ private function displaySqlParams(array $sql, bool $oneLine = false): string {
 
 		return '';
 	}
+
+
+	/**
+	 *
+	 */
+	private function forceDisableMaintenanceMode() {
+		$CONFIG = [];
+		$cwd = getcwd();
+		if (is_bool($cwd)) {
+			return;
+		}
+
+		$configFile = $cwd . '/config/config.php';
+		include $configFile;
+
+		if (empty($CONFIG) || $this->getBool(ConfigService::MAINTENANCE, $CONFIG) === true) {
+			return;
+		}
+
+		$CONFIG[ConfigService::MAINTENANCE] = false;
+		file_put_contents(
+			$configFile, '<?php' . "\n" . '$CONFIG = ' . var_export($CONFIG, true) . ';' . "\n"
+		);
+	}
```

Comment:
This seems odd, no?  🙈

@nextcloud/server-backend

Reply:
_yes it does :]_

my reason(s):

when running the restoring process, `config/config.php` might been modified:
- during the restoring process, the file could have been erased by the backup's version.
- after the restoration, if some path/sql params have been changed to fit the new setup, the occ command will have to update those values manually.

Now, I am not even sure that this current `config/config.php` is the one that have been extracted from the restoration of the backup, in case the admin has choosen another path for the extraction of the data.

Meaning, I cannot trust the values returned by `IConfig::getSystemValue()`, nor I want to use `IConfig::setSystemValue()` to update `config/config.php`.

**tl;dr**: If I remove the maintenance mode using `IConfig`, `config/config.php` will returns to its previous state from when I enabled the maintenance mode, and all modification done during the restoring process will be lost

Reply:
Ah yes, that make sense!
No idea if there is a more pleasant solution though :thinking:

Reply:
Seems reasonable considering the limitation of setSystemValue that it will take the cached config from when the php process hits the system config for the first time: https://github.com/nextcloud/server/blob/b78f3a57d1d05fb5f81ec8361139b64a3e54d7b0/lib/private/Config.php#L245

Maybe it makes sense to be on the safe side to sitill adapt the locking andpermission change logic from https://github.com/nextcloud/server/blob/b78f3a57d1d05fb5f81ec8361139b64a3e54d7b0/lib/private/Config.php#L243-L275

Or as an alternative idea, to reuse the existing config writing code but with the limitation that there is no public API it would be an option to create a new \OC\Config object as in https://github.com/nextcloud/server/blob/03936d776272e5bd270fb185ad7bcb438cc635d7/lib/base.php#L150 in order to reload the system config with the restored state

Reply:
> No idea if there is a more pleasant solution though

I came to the same conclusion than @juliushaertl, the alternate solution would be to add a method in `IConfig` to refresh the system config. Now, is it really something we need out of this use case ?

Reply:
Might be a better idea to 'force' the read/cache refresh of `config/config,php` within the method that write the _'new'_ config file